contracts/DCAStrategies/DCAStrategies/DCAStrategiesManagementHandler.sol

```diff
@@ -4,6 +4,12 @@ pragma solidity >=0.8.7 <0.9.0;
 import '../../interfaces/IDCAStrategies.sol';
 
 abstract contract DCAStrategiesManagementHandler is IDCAStrategiesManagementHandler {
+  modifier onlyStratOwner(uint80 _strategyId) {
```

Comment:
I really like the modifier, but you are reading `_strategies[_strategyId]` twice. Once here, and once to get the new version. That's one extra storage read, which we could avoid if we just do it "manually" inside `updateStrategyTokens`